types/reveal.js/index.d.ts

```diff
@@ -0,0 +1,1622 @@
+// Type definitions for Reveal 4.3
+// Project: https://github.com/hakimel/reveal.js/
+// Definitions by: robertop87 <https://github.com/robertop87>,
+//                 Nava2 <https://github.com/Nava2>,
+//                 JPtenBerge <https://github.com/JPtenBerge>
+//                 Keita Watanabe <https://github.com/kwatanwa17>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace Reveal;
+
+export = Reveal;
+
+// NOTE: this does not work when only `options` is passed
+// const Reveal: new (revealElement?: Element | null, options: Reveal.Options) => Reveal.Api;
+
+// possible solutions
+// 1. place optional parameters after the required parameter
+// 2. use object as parameter
+//
+// a possible workaround is not checking parameters
+
+/**
+ * reveal.js - MIT licensed
+ *
+ * Copyright (C) 2011-2022 Hakim El Hattab, https://hakim.se
+ *
+ * @see {@link https://revealjs.com}
+ * @see {@link https://github.com/hakimel/reveal.js/blob/master/js/reveal.js}
+ */
+declare const Reveal: {
+    new (...args: any): Reveal.Api;
+    initialize(options?: Reveal.Options): Promise<Reveal.Api>;
+};
```

Comment:
This can be accomplished via overloads:

```suggestion
/**
 * reveal.js - MIT licensed
 *
 * Copyright (C) 2011-2022 Hakim El Hattab, https://hakim.se
 *
 * @see {@link https://revealjs.com}
 * @see {@link https://github.com/hakimel/reveal.js/blob/master/js/reveal.js}
 */
declare const Reveal: {
    new (options?: Reveal.Options): Reveal.Api;
    new (revealElement: Element, options: Reveal.Options): Reveal.Api;
    initialize(options?: Reveal.Options): Promise<Reveal.Api>;
};
```

Also, it looks like `revealElement` is required if you invoke `Reveal` with two arguments: 
- https://github.com/hakimel/reveal.js/blob/master/js/reveal.js#L44 (only queries `.reveal` if `arguments.length` is less than 2)
- https://github.com/hakimel/reveal.js/blob/master/js/reveal.js#L124 (throws if `revealElement` isn't defined).